### DISCUSSION TITLE
Verify claims list support

### DISCUSSION THREAD

**Discussion by sj-lt** [emqx/emqx#5161] [Ideas]
Hello,
I am very glad that token verification via jwks is now implemented.
Is it possible to add claim verification as if it's in a list?
My scenario is that i have a claim in a token that hold multiple values. I would like to check if value defined in emqx configuration is there.
This could look like this in emqx_auth_jwt.erl :
```
do_verify_claims([{Key, Expected} | L], Claims) ->
    case do_verify_claim(maps:get(Key, Claims, undefined),Expected) of
        true -> do_verify_claims(L, Claims);
        false -> {error, {verify_claim_failed, Key}}
    end.

do_verify_claim(Value,Expected) when is_list(Value) -> 
    lists:any(fun(X) -> X =:= Expected end,Value);
do_verify_claim(Value,Expected) -> 
    Value =:= Expected.     
```
Thanks in advance

**Comment by tigercl** [comment]
@slutkiewicz  Sorry for not replying to you in time, can I know more about your usage scenario, I am not sure under what circumstances the value of claim will be a list, I want to know if this is necessary.

**Comment by sj-lt** [comment]
Hello I would like to verify aud field of my token. According to RFC 7519 https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3. 

> In the general case, the "aud" value is an array of case- sensitive strings, each containing a StringOrURI value. In the special case when the JWT has one audience, the "aud" value MAY be a single case-sensitive string containing a StringOrURI value

And in my case it is an array. Dunno if it is necessary for all other claims that you wish to verify. Maybe there could be some different mechanism.

**Comment by haayv** [comment]
I would like to bump this discussion as it has been lying dormant without any answer for 4+ years and i have the same issue. It seems it is still not possible to use the claim verification feature of the JWT auth to check the aud claim in the token. Have i missed something? Are there any plans regarding the support lists in the claim verification feature in general?

**Comment by zmstone** [comment]
Hi @haayv 

The requirement is clear, we just haven't had the chance to prioritize it yet.
I'll bring this up in the planning.

**Comment by zmstone** [comment]
Will add 'aud' claim verification in 6.0.2 and 6.1.1: 
PR: https://github.com/emqx/emqx/pull/16423
